### PR TITLE
Draft: support slot parsing inside dict and lists of dict or strings

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Adds support for parsing slots inside of dictionaries and dicts and strings inside of lists.
This enables slot support for subkeys

### What issues does this PR fix or reference?
#51798
#50596

### Previous Behavior
As mentioned in linked issues, slots were not parsed in subkeys of state files

The following states displays the slot text itself
```
slot test:
  file.managed:
    - name: /tmp/thing
    - source: salt://slot.jinja
    - template: jinja
    - context:
        custom_var: __slot__:salt:cmd.run('echo password')

test module stuff:
  module.run:
    - test.arg:
      - message: __slot__:salt:cmd.run('echo blah')

scheduled_slot:
  schedule.present:
    - function: test.arg
    - job_args:
        - __slot__:salt:cmd.run('echo "blah"')
    - cron: '* * * * *'

scheduled_slot2:
  schedule.present:
    - function: test.arg
    - job_kwargs:
        - message: __slot__:salt:cmd.run('echo "blah"')
    - cron: '* * * * *'
```
### New Behavior
Previously mentioned state examples now show slot results

### Tests written?
Yes

### Commits signed with GPG?
No

@DmitryKuzmenko

### CODE REVIEW

**Review by DmitryKuzmenko — APPROVED** [review]
I'm okay with this code.

**Comment by DmitryKuzmenko** [comment]
@thatch45 could you please take a look at this?

File: salt/state.py [hunk 1]
```diff
@@ -2120,18 +2120,53 @@ def format_slots(self, cdata):
...
             for ind, arg in avalues:
                 arg = salt.utils.data.decode(arg, keep=True)
-                if not isinstance(arg, six.text_type) \
-                        or not arg.startswith('__slot__:'):
+                if isinstance(arg, dict):
+                    # Search dictionary values for __slot__:
+                    for key, value in arg.items():
+                        try:
+                            if value.startswith(SLOT_TEXT):
+                                log.debug("Slot processsing dict value %s", value)
```
[the remark this answers is not in the document]
Comment (mchugh19) [reply salt/state.py]: Done!

**Comment by mchugh19** [comment]
Tests added

**Review by thatch45 — APPROVED** [review]

**Comment by thatch45** [comment]
For the record, this is fantastic and I wanted to get it in when I added slots but I just did not have the time. Is there anything blocking merging this?

**Comment by mchugh19** [comment]
As far as I know it's all set. There's also #52187 you might want to review as well.

**Pull request merged**